test/regress/regress0/arith/bug569.smt2

```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: --no-check-unsat-cores --no-check-proofs
+; COMMAND-LINE: --no-check-unsat-cores
```

Comment:
I suggest checking coverage of these tests w.r.t. new proof code to make sure that all new derivation rules are exercised. 
https://github.com/CVC4/CVC4/wiki/Developer-Guide#code-coverage-of-unitsystemregression-tests

Alternatively, you could manually generate and check the proof or temporarilly plant a bug. The reason I am suggesting this is that sometimes proofs pass because they are not even generated because of some flag that was turned on in the smt engine or something like that.

Reply:
Thanks! I've done this. Some paths are actually being missed. Specifically, I'm having a hard time getting the `IntReal` tightening rules to go off. I'll keep working on this.

Reply:
Update: I've realized that the rewriter purifies `IntReal` terms into `IntInt` terms, so the former kind of term will never occur in proofs. I've removed the code for the applicable rules, and have added assertions stating that indeed no `IntReal` terms occur in proofs.

I didn't modify that the term-printing code, because they can occur in the input, and must therefor be printable.

Reply:
Addition: I'll remove the axioms for `IntReal` in a later PR.
I'm not doing it now because this will require me to change some of the tests in `th_lira_test.plf`, and this PR is already big enough.